ServerCore/Helpers/SubmissionEvaluator.cs

```diff
@@ -0,0 +1,348 @@
+    /// </summary>
+    public class SubmissionView
+    {
+        public Submission Submission { get; set; }
+        public Response Response { get; set; }
+        public string SubmitterName { get; set; }
+        public bool IsFreeform { get; set; }
+        public string FreeformReponse { get; set; }
+    }
+
+    /// <summary>
+    /// Handles player submissions
+    /// </summary>
+    public class SubmissionEvaluator
+    {
+        /// <summary>
+        /// Evaulates player submissions then either saves them to the database or returns an error to the caller
+        /// </summary>
+        public static async Task<SubmissionResponse> EvaluateSubmission(PuzzleServerContext context, PuzzleUser loggedInUser, Event thisEvent, int puzzleId, string submissionText, bool allowFreeformSharing)
+        {
+            //Query data needed to process submission
+            Team team = await UserEventHelper.GetTeamForPlayer(context, thisEvent, loggedInUser);
+
+            Puzzle puzzle = await context.Puzzles.Where(
+                (p) => p.ID == puzzleId).FirstOrDefaultAsync();
+
+            PuzzleStatePerTeam puzzleState = await (PuzzleStateHelper
+                .GetFullReadOnlyQuery(
+                    context,
+                    thisEvent,
+                    puzzle,
+                    team))
+                .FirstAsync();
+
+            List<SubmissionView> submissionViews = await (from sub in context.Submissions
+                                                          join user in context.PuzzleUsers on sub.Submitter equals user
+                                                          join r in context.Responses on sub.Response equals r into responses
+                                                          from response in responses.DefaultIfEmpty()
+                                                          where sub.Team == team &&
+                                                          sub.Puzzle == puzzle
+                                                          orderby sub.TimeSubmitted
+                                                          select new SubmissionView()
+                                                          {
+                                                              Submission = sub,
+                                                              Response = response,
+                                                              SubmitterName = user.Name,
+                                                              FreeformReponse = sub.FreeformResponse,
+                                                              IsFreeform = puzzle.IsFreeform
+                                                          }).ToListAsync();
+
+            List<Submission> submissions = new List<Submission>(submissionViews.Count);
+            foreach (SubmissionView submissionView in submissionViews)
+            {
+                submissions.Add(submissionView.Submission);

```

Comment:
Why not just get the submissions directly if this is all you want? submissionViews is unused beyond this point.

Reply:
That's something I'll address in post-event clean up, I wasn't confident enough in my sql syntax to rewrite the queries.